helm/prometheus-rules/templates/alerting-rules/node.workload_cluster.rules.yml

```
@@ -125,3 +125,14 @@ spec:
         team: {{ include "providerTeam" . }}
         topic: kubernetes
     {{- end }}
+    - alert: WorkloadClusterMasterMemoryUsageTooHigh
+      annotations:
+        description: '{{`Machine {{ $labels.instance }} memory usage is too high (less than 10% and 2G of allocatable memory).`}}'
+      expr: ((((node_memory_MemTotal_bytes{cluster_type="workload_cluster"} - node_memory_MemFree_bytes{cluster_type="workload_cluster"} - node_memory_Cached_bytes{cluster_type="workload_cluster"}) / (node_memory_MemTotal_bytes{cluster_type="workload_cluster"}) * 100)) > 90 and (node_memory_MemFree_bytes{cluster_type="workload_cluster"} + node_memory_Cached_bytes{cluster_type="workload_cluster"}) < 2147483648) and on (node) kube_node_role{role="master"}
```

Writing the expression on multiple lines helps a lot with reading it!

```suggestion
      expr: |-
        ( ( (
              ( node_memory_MemTotal_bytes{cluster_type="workload_cluster"}
              - node_memory_MemFree_bytes{cluster_type="workload_cluster"}
              - node_memory_Cached_bytes{cluster_type="workload_cluster"}
              )
              / (node_memory_MemTotal_bytes{cluster_type="workload_cluster"}
            ) * 100
          )
        ) > 90
        and
        ( node_memory_MemFree_bytes{cluster_type="workload_cluster"}
          + node_memory_Cached_bytes{cluster_type="workload_cluster"}
        ) < 2147483648) 
        and on (node) kube_node_role{role="master"}
```